spec/requests/api/graphql/graphql_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+require 'rails_helper'
+
+RSpec.describe API::GraphqlController, issues: ['railgun#147'] do
+
+  context 'Administrator can use users and user_roles' do
+    let(:authenticated_header) {
+      { 'Authorization' => "Bearer #{create(:administrator).api_token}" }
+    }
+
+    describe 'post /api/graphql#execute' do
+      it 'valid request returns users' do
+        users = create_list(:user, 2)
+        post '/api/graphql', headers: authenticated_header, params: { query: '{ users { id email }}' }
+
+        expect(response.body).to match(users.first.email)
+      end
+
+      it 'valid request returns user' do
+        user = create(:user)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ user(id: \"#{user.id}\") { id email }}" }
+
+        expect(response.body).to match(user.email)
+      end
+
+      it 'valid request returns user_roles' do
+        user = create(:owner)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ user_role(user_id: #{user.id}) { roles }}" }
+
+        expect(response.body).to match('owner')
+      end
+    end
+  end
+
+  context 'Developer can use snippets' do
+    let(:authenticated_header) {
+      { 'Authorization' => "Bearer #{create(:developer).api_token}" }
+    }
+
+    describe 'post /api/graphql#execute' do
+      it 'valid request returns snippets' do
+        snippets = create_list(:snippet, 2)
+        post '/api/graphql', headers: authenticated_header, params: { query: '{ snippets { slug }}' }
+
+        expect(response.body).to match(snippets.first.slug)
+      end
+
+      it 'valid request returns snippets' do
+        snippet = create(:snippet)
+        post '/api/graphql', headers: authenticated_header, params: { query: "{ snippet(slug: \"#{snippet.slug}\") { slug text }}" }
+
+        expect(response.body).to match(snippet.text)
```

Comment:
better to parse body and check actual returned object, not only search for the text part